syntax = "proto3";
import "common.proto";
option go_package = "./comms/container";

// The greeting service definition.
service VolpeContainer {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc InitFromSeed (Seed) returns (Reply) {}

  rpc InitFromSeedPopulation (Population) returns (Reply) {}
  // used to init or incorporate a population

  rpc GetBestPopulation (PopulationSize) returns (Population) {}

  rpc AdjustPopulationSize (PopulationSize) returns (Reply) {}

  rpc RunForGenerations (PopulationSize) returns (Reply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

message Seed {
  int32 seed = 1;
}

message PopulationSize {
  int32 size = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message Reply {
  bool success = 1;
  string message = 2;
}
